### PR TITLE
Fix transaction plan when using max amount

### PR DESCRIPTION
**Pull request by BSathvik**
## Description

Fix `TransactionBuilder::plan` such that it does not add uncessary change output to the transaction when sending max amount 

## Checklist

- [ ] Prefix PR title with `[WIP]` if necessary.
- [ ] Add tests to cover changes as needed.
- [ ] Update documentation as needed.

### CODE REVIEW

File: src/Bitcoin/TransactionBuilder.h [hunk 1]
```diff
@@ -22,13 +22,22 @@ struct TransactionBuilder {
...
         plan.amount = input.amount();
 
+        auto output_size = 2;
+
         if (input.use_max_amount() && UnspentSelector::sum(input.utxo()) == plan.amount) {
-            plan.amount -= UnspentSelector::calculateFee(input.utxo().size(), 2, input.byte_fee());
+            output_size = 1;
+
+            auto input_size = 0;
+            for(auto utxo: input.utxo()) {
```
Comment (alejandro-isaza) [inline src/Bitcoin/TransactionBuilder.h]: Space after `for` please

File: src/Bitcoin/TransactionBuilder.h [hunk 2]
```diff
@@ -22,13 +22,22 @@ struct TransactionBuilder {
...
+                if (utxo.amount() > UnspentSelector::calculateSingleInputFee(input.byte_fee())) { input_size++; }
```
Comment (alejandro-isaza) [inline src/Bitcoin/TransactionBuilder.h]: New line for if contents please

**Comment by alejandro-isaza** [comment]
Let's add a test case as @hewigovens suggests

**Comment by hewigovens** [comment]
This pr is more than 10 days old, @BSathvik any plan to work on? should we close it ?

File: tests/Bitcoin/TransactionPlanTests.cpp [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+#include "Bitcoin/TransactionPlan.h"
+#include "Bitcoin/TransactionBuilder.h"
+#include "proto/Bitcoin.pb.h"
+
+#include <gtest/gtest.h>
+
+using namespace TW;
+using namespace TW::Bitcoin;
+
+auto txOutPoint = OutPoint(std::vector<uint8_t>(32), 0);
```
Comment (alejandro-isaza) [inline tests/Bitcoin/TransactionPlanTests.cpp]: let's make this `const` so that it's not accidentally modified by a test

**Comment by hewigovens** [comment]
@BSathvik let's merge this once iOS CI passed

**Pull request merged**